### DISCUSSION TITLE
[FAIL. Reason: panic: arithmetic underflow or overflow (0x11)] testSwapBuyEth()

### DISCUSSION THREAD

**Discussion by cosmosito** [Jeiwan/uniswapv3-book#95] [Milestone 1. First Swap]
Fails after calling this function:
UniswapV3PoolTest::uniswapV3SwapCallback(-8396714242162444 [-8.396e15], 42000000000000000000 [4.2e19])

    function uniswapV3SwapCallback(int256 amount0, int256 amount1) public {
        if (amount0 > 0) {
            token0.approve(msg.sender, uint256(amount0));
            token0.transferFrom(address(this), msg.sender, uint256(amount0));
        }

        if (amount1 > 0) {
            token1.approve(msg.sender, uint256(amount1));
            token1.transferFrom(address(this), msg.sender, uint256(amount1));
        }
    }

**Comment by vishwa7dcap786** [comment]
Make sure that you have provided enough allowence for amount1
